app/helpers/govuk_rails_compatibile_link_helper.rb

```diff
@@ -0,0 +1,157 @@
+require "html_attributes_utils"
+
+module GovukRailsCompatibileLinkHelper
+  using HTMLAttributesUtils
+
+  def govuk_link_classes(*styles, default_class: "#{brand}-link")
+    if (invalid_styles = (styles - link_styles.keys)) && invalid_styles.any?
+      fail(ArgumentError, "invalid styles #{invalid_styles.to_sentence}. Valid styles are #{link_styles.keys.to_sentence}")
+    end
+
+    [default_class] + link_styles.values_at(*styles).compact
+  end
+
+  def govuk_button_classes(*styles, default_class: "#{brand}-button")
+    if (invalid_styles = (styles - button_styles.keys)) && invalid_styles.any?
+      fail(ArgumentError, "invalid styles #{invalid_styles.to_sentence}. Valid styles are #{button_styles.keys.to_sentence}")
+    end
+
+    [default_class] + button_styles.values_at(*styles).compact
+  end
+
+  def govuk_link_to(name = nil, options = nil, extra_options = {}, &block)
+    extra_options = options if block_given?
+    html_options = build_html_options(extra_options)
+
+    if block_given?
+      link_to(name, html_options, &block)
+    else
+      link_to(name, options, html_options)
+    end
```

Comment:
I think you can avoid the `if` conditions here and always pass `block` on to the rails helpers, as those do their own `if block_given?` checks?

```suggestion
    extra_options = options if block_given?
    html_options = build_html_options(extra_options) || nil

    link_to(name, options, html_options, &block)
```

Reply:
Going to leave these as they're in the legacy implementation which will soon be dropped.